### PR TITLE
[stable/minio] Allow providing a list of trusted certificates from Kubernetes secrets

### PR DESCRIPTION
**Pull request by tacrocha**
Signed-off-by: Tacio Rocha <tacio.rocha@guavus.com>

#### Special notes for your reviewer:

This PR refers to this issue: https://github.com/minio/charts/issues/15

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/minio]`)

### CODE REVIEW

File: minio/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: High Performance, Kubernetes Native Object Storage
 name: minio
-version: 6.1.1
+version: 6.1.2
```
Comment (harshavardhana) [inline minio/Chart.yaml:4]: the value has changed please update with the master @tacrocha

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Does it not conflict with existing paths for `CAs/public.crt` elsewhere in the code base?

**Comment by tacrocha** [comment]
> Does it not conflict with existing paths for `CAs/public.crt` elsewhere in the code base?

Yes, indeed. I tried and couldn't find a solution for having Minio's own public certificate together with other trusted certs, in the same mountPath, using only Kubernetes volumes and secrets. I have some options: 

1. Mount each trusted certificate on its own subdirectory of `{{ .Values.certsPath }}/CAs`. That would require Minio to recurse into subdirectories during initialization. I haven't inspected Minio's source code, but, in my tests, that did not work.
2. Use an `initContainer` to copy all the certificates mounted individually from secrets into an `emptyDir`, then mount that `emptyDir` at `{{ .Values.certsPath }}/CAs`
3. Mount the trusted certificates separately, from a secret that may contain Minio's public certificate combined with other trusted certificates. I have tested that myself by changing the `StatefulSet` directly.

Should I close this PR and bring this discussion to issue [#15](https://github.com/minio/charts/issues/15)?

**Comment by tacrocha** [comment]
@harshavardhana I've updated my PR to allow the tls keys and trusted certs to coexist, by making the trusted certs a separate volume. If TLS is enabled and `trustedCerts` is empty, the chart will behave just like before and place the `public.crt` inside the `CAs` directory. If TLS is disabled and you need to trust a 3rd party certificate, you may provide a secret which contains as many certificates as needed and it'll be mounted on the `CAs` directory. If you need both TLS enabled and trusted certificates, you may simply include minios's public cert in the secret with the key being `public.crt`.

**Comment by harshavardhana** [comment]
> @harshavardhana I've updated my PR to allow the tls keys and trusted certs to coexist, by making the trusted certs a separate volume. If TLS is enabled and `trustedCerts` is empty, the chart will behave just like before and place the `public.crt` inside the `CAs` directory. If TLS is disabled and you need to trust a 3rd party certificate, you may provide a secret which contains as many certificates as needed and it'll be mounted on the `CAs` directory. If you need both TLS enabled and trusted certificates, you may simply include minios's public cert in the secret with the key being `public.crt`.

@nitisht can you check this?

**Review by nitisht — CHANGES_REQUESTED** [review]
changes look good @tacrocha . Do you mind adding documentation about this in the README.md please?

**Comment by harshavardhana** [comment]
Can update the chart to be 7.1.0? @tacrocha

**Comment by tacrocha** [comment]
Hi. I've documented the new parameter in the `README` updated the chart version.

**Comment by harshavardhana** [comment]
@tacrocha can you update the chart version?

**Comment by tacrocha** [comment]
Chart version was updated.

**Review by harshavardhana — APPROVED** [review]

**Review by nitisht — APPROVED** [review]
LGTM, thanks @tacrocha

**Pull request merged**